README.md

```diff
@@ -38,13 +38,24 @@ You should see output that looks something like this:
 
 ### Nuke
 
-Add a reference to `Nuke.OctoVersion` in your `_build` project.
-Add a parameter marked with `[NukeOctoVersion]` and it will be auto-populated:
+Add a reference to `OctoVersion.Tool` in your Nuke `_build` project. This can be done using the nuke command:
+
+```
+nuke :add-package OctoVersion.Tool --version X.X.X
+```
+
+Add a parameter marked with `[OctoVersion]` and it will be auto-populated:
 
 ```c#
-[NukeOctoVersion] readonly OctoVersionInfo OctoVersionInfo;
+// The Required Attribute will automatically throw an exception if the 
+// OctoVersionInfo parameter is not set due to an error or misconfiguration in Nuke.
+[Required]
+[OctoVersion(AutoDetectBranchName = true)]
```

Comment:
We don't want to recommend this - see [this part](https://github.com/OctopusDeploy/OctoVersion#cant-you-figure-out-the-branch-im-on-why-do-i-have-to-specify-it) of the readme.

The approach I've used in a bunch of places is:
```
    [Parameter("Branch name for OctoVersion to use to calculate the version number. Can be set via the environment variable OCTOVERSION_CurrentBranch.", Name = "OCTOVERSION_CurrentBranch")] readonly string BranchName;
    [Parameter("Whether to auto-detect the branch name - this is okay for a local build, but should not be used under CI.")] readonly bool AutoDetectBranch = IsLocalBuild;
    [OctoVersion(UpdateBuildNumber = true, BranchParameter = nameof(BranchName), AutoDetectBranchParameter = nameof(AutoDetectBranch), Framework = "net6.0")] readonly OctoVersionInfo OctoVersionInfo;
```

(the `Framework = "net6.0"` is only required if your app is net6)

Reply:
Thanks for that, yeah I wasn't sure how many/which properties to show for the example. I've changed it to your suggestion. Does it look ok?